apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/TransactionDaoImpl.java

```diff
@@ -372,22 +375,38 @@ public synchronized Long getTransactionCount(TransactionalDataSource dataSource,
 
         StringBuilder buf = new StringBuilder();
         buf.append("SELECT transaction.* FROM transaction ");
-        createTransactionSelectSql(buf, "transaction.*", accountId, numberOfConfirmations, type, subtype, blockTimestamp, withMessage, phasedOnly, nonPhasedOnly, includeExpiredPrunable, executedOnly, includePrivate, height, prunableExpiration);
-        buf.append(DbUtils.limitsClause(from, to));
+        createTransactionSelectSql(buf, "transaction.*", accountId, numberOfConfirmations, type, subtype,
+                blockTimestamp, withMessage, phasedOnly, nonPhasedOnly, includeExpiredPrunable, executedOnly, includePrivate, height, prunableExpiration);
+        // condition for main or shard db
+        if (dataSource.getDbIdentity().isEmpty()) {
+            buf.append(DbUtils.limitsClause(from, to)); // main db
+        } else {
+            buf.append(DbUtils.limitsClauseInShardDb(from, to)); // shard db
```

Comment:
Is it really necessary?

Reply:
Yes, because offset value is not correct anymore.